packages/ampd-sdk/src/grpc/connection_pool.rs

```diff
@@ -145,39 +216,31 @@ impl ConnectionPool {
 
         match endpoint.connect().await {
             Ok(channel) => {
-                self.notify_clients(ConnectionState::Connected(channel), false)?;
+                self.notify_clients(ConnectionState::Connected(channel));
```

Comment:
it's a bit confusing to have success notifications here and failure notifications outside, could you move this one to the `handle_connection_failure` function?. Also, when possible try to use `map` and `inspect` and their `_err` equivalent for these kind of cases instead of matching on results